### ISSUE TITLE
Incompatible with Thunderbird 115 and test versions

### ISSUE THREAD

**Issue by pmruzicka** [xnotepp/xnote#113]
Is it safe to assume this add-on is dead. I just opened my TBird and it was updated to 115. XNote++ now incompatible. Used this for years. So I am not sure I can add this as an issue. Version I see as latest is 3.3.2 July 2, 2022. Is there a version in testing I can download and try? Does anyone read these anymore?

**Comment by opto** [comment]
if you look at addons.thunderbird.net (https://addons.thunderbird.net/de/thunderbird/addon/xnotepp/), you will see that the latest version 3.5.14  is of July 3rd, 2023. So of course it will be updated.

Note: if you should have Kaspersky installed (maybe also some other antivirus), some of them block the update server for addons, while main TB updates are possible.

**Comment by zevrix** [comment]
XNote 3.5.14 update notes inform that Thunderbird 115+ doesn't allow custom columns (this message was displayed when I updated, but I don't know how to get to this message anymore). The message also mentions that TB developers promise to restore this feature (custom columns) at some point.

If XNote's custom column cannot be enabled, this means that XNote is at this point is pretty much dead with 115+, unfortunately. XNote is the main reason I'm using TB as it makes email management so much easier.

I wonder if there's any info on when TB will allow custom columns again?

Until then I'm reverting to TB 104 so I can use XNote as usually.

**Comment by opto** [comment]
oh, there are 'many' possibilities.
In any case, you can use the XNote tag.

As all is html now, one could see whether one can add columns without the add column mechanism of TB. But it would be fixed position, then.
Or one could try to have the star icon double colored, showing star and xnote in one icon.

I was told it would come in one of the next versions. Currently, TB 115 is not yet distributed officially (users are not updated), so I decided to wait for the TB team to add this ... for now.

Klaus

**Comment by zevrix** [comment]
Thanks for the info, Klaus, and for your great work.

To me, the custom column is THE most important feature of XNote. It's what makes XNote amazing, and what in turn makes TB better than other email clients.

This column makes all the difference between chaos and order in the inbox (and other email folders).

I attached a mockup that illustrates why XNote column is so important. You can briefly annotate what's needed to do regarding certain emails, then just glimpse through the list without having to select and read each individual email over and over again.

![image](https://github.com/xnotepp/xnote/assets/51013144/f59190d3-33c1-4d00-a0a2-aecf533d161c)

**Comment by rlaggren** [comment]
For those not needing/using any buggy features is there any reasonable way to force xnote install on TB-115?  Eg. any dev versions that mostly run?

Rufus

**Comment by opto** [comment]
yes, that will be coming as alpha or beta version "soon"

**Comment by pmruzicka** [comment]
How soon?

**Comment by cebu23** [comment]
Thunderbird is now starting to gently nag me to update to 115.2 - I'm currently on 102.15.0 and the "About Mozilla Thunderbird" dialog has the "Update to 115.2.0" prominently where I can hit it by mistake.  I know compatibility work is still underway for Xnote - wondering if there might be a way to release an interim version or fork that would work for 115 even if all functionality isn't present (i.e. protect users who screw up and upgrade from losing ALL functionality, even if they lose some like the custom column).

I will studiously stay AWAY from the upgrade button for now.  :-)

**Comment by jeffgoh** [comment]
I see 3.5.16 but my tbird 115.1.0 steadfastly refuses to install it.

**Comment by GerryT11** [comment]
FYI: Thunderbird auto-updates to 115.2.3 (64-bit) seem to have started. I tried to avoid it (blocked version), but now it has automatically updated. Now, at the moment, without functioning XNote++

**Comment by opto** [comment]
this is a first test version for 115. Not for production use

Known issues:

TB has not yet implemented custom columns
print not working
set tag makes XNote button inoperative

Please report further issues here.

Thanks,

Klaus

**Comment by opto** [comment]
actually, here it is: https://github.com/xnotepp/xnote/releases/tag/4.0.1

**Comment by gjawans** [comment]
Installed but I don't see a single my note, not even the option in the menu to add new one. (TB 115.2.3, PL)

**Comment by StefanPp** [comment]
For me it work on 115.3.1 (64-Bit Windows 10).
Symbols in the Pop-Up-Note looking shuffled.
But opening Notes via Menubar works, saving an re-opening on click works fine. Thanks a lot!!

**Comment by pmruzicka** [comment]
Are all the old notes gone, or is there a setting that needs to be turned on to have them show up? I did figure out how to add new ones. And the symbols/icons at the bottom of the note are garbled. Also, there seems to be no way to move a note on the screen. One more thing, the donation page pops up but there is no link to donate. Where can I do that?

**Comment by opto** [comment]
https://github.com/xnotepp/xnote/releases/tag/v4.0.2

this version sets the XNote tag, if that is enabled in options. Currently, there is no custom column yet in TB, so that is the only way to show which mails have a XNote (other than search in general search or with expression search addon - did not test whether tht communication works in 115, and there is a test version for that in github as well.

Donations: many thanks for considering. I will look at the popup, but the link here works: https://addons.thunderbird.net/de/thunderbird/addon/xnotepp/?src=ss

**Comment by opto** [comment]
printing not working yet, and the display inside the message is not working yet

**Comment by opto** [comment]
this version shows the note in the message display and in the message window

https://github.com/xnotepp/xnote/releases/tag/v4.0.3

**Comment by KL7000F** [comment]
Unfortunately, I've now been forced to update to Thunderbird 115.3.2 :( I'm glad you're currently putting a lot of work into a new XNote version. I'll make sure I donate some to you too! Great addon! I'm in the trial version noticed the following bugs.

Bugs version 4.0.3:

- XNote window cannot be moved.
- Position information in the settings has no effect.
- XNote within the email should optionally always be opened.
- Missing column with symbol in the overview.
- If necessary, a column optionally with text content of the XNote.
- XNote is missing from the context menu.

**Comment by opto** [comment]
@KL7000F 
>XNote within the email should optionally always be opened.  
:can be set in the options
>Position information in the settings has no effect.  
:currently only for new notes

In the next days, I will upload this version to have at least some note support.

@pmruzicka 
>Are all the old notes gone?  
:no, all notes need to be in the folder that is set in options.

**Comment by KL7000F** [comment]
@opto 

> :can be set in the options

Yes, but it has no effect. The XNote within the email is always displayed "small" and must be enlarged manually. Even with newly created XNote's.

**Comment by opto** [comment]
@KL7000F I think that never worked, but it is fixed now for the next release

**Comment by opto** [comment]
this version allows to move the note, use the context menue in threadtree and have a full disply of the note in the message:

https://github.com/xnotepp/xnote/releases/tag/v4.0.4

**Comment by jnagler** [comment]
Hi Klaus, thanks for all your effort! Unfortunately my time is too limited to do a thorough report. Where should I post my lay experiences with your v4.0.4? I needed to go back to v4.0.3 as XNote was not working. There was no window to create/edit a note. Existing notes displayed okay on top of the email body. The list of emails (sorry, I am not experienced in the terms to better describe the Thunderbird GUI parts) was altered and space was added to the right of the "Select columns to display" part with a horizontal scroll bar now appearing for this GUI part. The context menu was now more working and still is not after downgrading. Best, Jürgen

**Comment by KL7000F** [comment]
In version 4.1.5 no keywords are written when a note is created.
An icon (or column) for notes now appears in the message overview, but this has no effect yet.

**Comment by jnagler** [comment]
Windows 11, TB 115.4.1, XNote++ 4.1.5: Thanks for the updates. Context menu is back Create window appears, position and size are based on the size and position settings, it's resizable and can be moved.

Things newly found:
- When opening (new or existing) its height is always too less to show the tools:
![image](https://github.com/xnotepp/xnote/assets/4779454/31b42c7a-2748-45e0-8721-df55d6c51386)
- The first tool icon shows an unclear hover text:
![image](https://github.com/xnotepp/xnote/assets/4779454/48adf550-49d9-4c49-9e39-a85397e2c737)
- Resizing/Repositioning is only stored if content of the note is also changed.
- Context menu is not always aware of a note already exists (not uniquely reproducable) and show "Create note" instead of modify/delete.
- I still see vertical space added to the mail list with now a horizontal scroll bar been added. I assume this may the icon/column that @KL7000F mentioned but on my Classic View it's just space, no icon or distinct column, nothing in the "Select columns to display" that may explain it.

Best, Jürgen

**Comment by opto** [comment]
custom columns are not yet implemented by TB. I will check whether we might be able to hijack an unimportant column (maybe size) and insert the notes there.

The newest version (coming today) can do keywords and other bugs are fixed

**Comment by opto** [comment]
this has a lot of fixes (but some from the screenshots are still missing ...

https://github.com/xnotepp/xnote/releases/tag/v4.1.7

**Comment by KL7000F** [comment]
Version 4.1.8:

- "Show XNote after email selection" is disabled, but the window still opens when I have marked an email with a note and then switch back from the calendar to the overview, for example.
- "Show XNote fully in the email" does not work.

Thanks for your work!

**Comment by opto** [comment]
this is a workaround that shows XNote in location column. See text in release here: please try this https://github.com/xnotepp/xnote/releases/tag/v4.2.0

**Comment by Eccenux** [comment]
Just wanted to note that the new API for columns in Thunderbird finally seems to come to life and it seems to be looking good 🙂
> Here is a proposal for an API design using the new methods implemented in [bug 1817682](https://bugzilla.mozilla.org/show_bug.cgi?id=1817682)
> https://docs.google.com/document/d/1uF-Z-kNBR3OVa_a2gQMRdnUMlHK67VDiofgO4hW8U2s/edit?usp=sharing
> Edit: Hit the "request access" button to receive commenting capabilities
